internal/app/repository/SQLStorage.go

```diff
@@ -0,0 +1,138 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	_ "github.com/jackc/pgx/v5/stdlib"
+)
+
+var (
+	ErrURLAlreadyExists = errors.New("url already exists")
+)
+
+func newErrURLAlreadyExists() error {
+	return ErrURLAlreadyExists
+}
+
+type SQLStorage struct {
+	DB   *sql.DB
+	cnfg string
+}
+
+func NewDB(cnfg string) *SQLStorage {
```

Comment:
```suggestion
func NewDB(cnfg string) (*SQLStorage, error) {
```
Если при инициализации стораджа произошла ошибка, необходимо либо сразу падать, не запуская сервер, либо откатываться на другую модель хранения данных. Попытка работать с неживым стораджем приведет лишь к массовым ошибкам в ответах и логах.